### ISSUE TITLE
AI and broken logic cause infinite loop and freeze the game with server load (100% CPU)

### ISSUE THREAD

**Issue by joepayne2493** [magefree/mage#5023]
Hi, I've never had any issues before, but II'm getting an error message when i want to login saying that the server is not responding.

I haven't changed any settings etc, so is it something on my end or are you guys doing maintenance?

Cheers for the hard work guys!

**Comment by LevelX2** [comment]
There does currently a bug exist, that if active, uses the cpu completly. Probably by endless iterating.
If this happens, the server does no longer respond to sign on tries and also it can't no longer handle active games. If the server is rest, it runs again until the same situation happens again (probably by someone using a specific card/action).

I'm currently trying to narrow down the causing cards/methods.

![grafik](https://user-images.githubusercontent.com/2066790/41213434-bd21baa6-6d45-11e8-907c-05507b7d9903.png)

**Comment by LevelX2** [comment]
![cpu01](https://user-images.githubusercontent.com/2066790/41510803-2e92a57c-726b-11e8-8879-e139c93b6c4f.jpg)
![cpu02](https://user-images.githubusercontent.com/2066790/41510804-2eb2c3e8-726b-11e8-9497-9e7a63a1be62.jpg)
![cpu03](https://user-images.githubusercontent.com/2066790/41510805-2ed1ac04-726b-11e8-8cac-93d29fa08f50.jpg)
![cpu04](https://user-images.githubusercontent.com/2066790/41510806-2eedd938-726b-11e8-8c07-ad8534630db2.jpg)
![cpu05](https://user-images.githubusercontent.com/2066790/41510807-2f1144a4-726b-11e8-91fe-fe85367bb61f.jpg)
![cpu06](https://user-images.githubusercontent.com/2066790/41510808-2f307d24-726b-11e8-86cd-a00f3cf1cc6a.jpg)

**Comment by LevelX2** [comment]
![cpu07](https://user-images.githubusercontent.com/2066790/41550518-48eb429e-7329-11e8-85f9-3456099cbad6.jpg)

**Comment by LevelX2** [comment]
![manaoptions](https://user-images.githubusercontent.com/2066790/41820899-429031d4-77d9-11e8-805e-fe360a8f3cea.jpg)

**Comment by LevelX2** [comment]
Here a java flight recorder file that shows exactly how the bug eats up the memory and blocks the server by garbage collection.

https://drive.google.com/file/d/1Jm96TKO9XvF9tyIXHa3FfO57mJe0jnD5/view?usp=sharing

**Comment by JayDi85** [comment]
Today I've checked de-server 2 hours ago and memory was fine (~30% used), but after 2 hours it ate all the memory (~85%) and down... Maybe someone used rare broken card with infinite loop... Hmm.

**Comment by Zerrisx** [comment]
Is this still an ongoing issue?

**Comment by JayDi85** [comment]
Yes, it's still true for some bad/broken cards. From user side that's bug like "game freezes with computer opponent".
